code/datums/components/toggled_clothing.dm

```diff
@@ -0,0 +1,203 @@
+/**
+ * Component which allows clothing to deploy a different kind of clothing onto you.
+ * The simplest example is hooded suits deploying hoods onto your head.
+ */
+/datum/component/toggled_clothing
+	/// Instance of the item we're creating
+	var/obj/item/clothing/deployable
+	/// Action used to toggle deployment
+	var/datum/action/item_action/toggle_action
+	/// Typepath of what we're creating
+	var/deployable_type
+	/// Which slot this item equips into
+	var/equipped_slot
+	/// Name of toggle action
+	var/action_name = ""
+	/// If true, we delete our deployable on toggle rather than putting it in nullspace
+	var/destroy_on_removal
+	/// Current state of our deployable equipment
+	var/currently_deployed = FALSE
+	/// What should be added to the end of the parent icon state when equipment is deployed? Set to "" for no change
+	var/parent_icon_state_suffix = ""
+	/// Icon state for overlay to display over the parent item while deployable item is not deployed
+	var/down_overlay_state_suffix = ""
+	/// Overlay to display over the parent item while deployable item is not deployed
+	var/mutable_appearance/undeployed_overlay
+	/// Optional callback triggered before deploying, return TRUE to continue or FALSE to cancel
+	var/datum/callback/pre_creation_check
+	/// Optional callback triggered when we create our deployable equipment
+	var/datum/callback/on_created
+	/// Optional callback triggered when we have deployed our equipment
+	var/datum/callback/on_deployed
+	/// Optional callback triggered before we hide our equipment, before as we may delete it afterwards
+	var/datum/callback/on_removed
+
+/datum/component/toggled_clothing/Initialize(
+	deployable_type,
+	equipped_slot,
+	action_name = "Toggle",
+	destroy_on_removal = FALSE,
+	parent_icon_state_suffix = "",
+	down_overlay_state_suffix = "",
+	datum/callback/pre_creation_check,
+	datum/callback/on_created,
+	datum/callback/on_deployed,
+	datum/callback/on_removed,
+)
+	. = ..()
+	if (!isclothing(parent))
+		return COMPONENT_INCOMPATIBLE
```

Comment:
If you don't check any clothing vars directly, this can be `isitem`. We have many equippable non-clothing items, would be neat to extend this functionality a bit. (Like a bedsheet that you can flip over your head or something.)